### PR TITLE
Update support for avaya-vsp and fix problem in functions.php for avaya-vsp

### PR DESCRIPTION
**Pull request by ospfbgp**
#### Please note

> Please read this information carefully.

- [x] Have you signed the [Contributors agreement](http://docs.librenms.org/General/Contributing/)
- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

Added to allow Avaya VSP switches to discover its ports.  All the Avaya VSP switches contain "Avaya Virtual Services Platform" in the port ifDescr and so you can not graph the ports with the current code.

### CODE REVIEW

**Comment by laf** [comment]
Will check through but yes this is how we wanted it :)

**Comment by ospfbgp** [comment]
I have updated all the files and put this in 1 pull request.  thanks for the patience I am new to github and librenms.

File: includes/definitions.inc.php [hunk 1]
```diff
@@ -1019,7 +1019,7 @@
 $config['os'][$os]['icon']             = 'avaya';
 
 $os = 'avaya-vsp';
-$config['os'][$os]['text']             = 'Avaya VSP';
+$config['os'][$os]['text']             = 'VOSS';
```
Comment (laf) [inline includes/definitions.inc.php]: How come this is changed to VOSS? It's only a text description and Avaya VSP seems more relevant still.

File: includes/discovery/os/avaya-vsp.inc.php [hunk 2]
```diff
@@ -2,7 +2,7 @@
 /*
  * LibreNMS
  *
- * Copyright (c) 2016 Neil Lathwood <neil@lathwood.co.uk>
```
Comment (laf) [inline includes/discovery/os/avaya-vsp.inc.php]: You can't remove my copyright, it's fine to add your copyright with it as well but you'll need to restore this line.

File: includes/polling/mempools/avaya-vsp.inc.php [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+// rcKhiSlotMemUsed 1.3.6.1.4.1.2272.1.85.10.1.1.6.1
+// rcKhiSlotMemFree 1.3.6.1.4.1.2272.1.85.10.1.1.7.1
+// rcKhiSlotMemUtil 1.3.6.1.4.1.2272.1.85.10.1.1.8.1
+
+if ($device['os'] == 'avaya-vsp') {
+    $used = (snmp_get($device, "1.3.6.1.4.1.2272.1.85.10.1.1.6.1", '-OvQ') * 1000);
+    $free = (snmp_get($device, "1.3.6.1.4.1.2272.1.85.10.1.1.7.1", '-OvQ') * 1000);
+    $perc = snmp_get($device, "1.3.6.1.4.1.2272.1.85.10.1.1.8.1", '-OvQ');
+    $total = ($used + $free);
+    echo "Avaya VSP Memory ";
```
Comment (laf) [inline includes/polling/mempools/avaya-vsp.inc.php]: You will need to change echo to d_echo(); I.e:

`d_echo("Avaya VSP Memory");`

File: includes/polling/os/avaya-vsp.inc.php [hunk 4]
```diff
@@ -2,15 +2,29 @@
 /*
  * LibreNMS
  *
- * Copyright (c) 2016 Neil Lathwood <neil@lathwood.co.uk>
```
Comment (laf) [inline includes/polling/os/avaya-vsp.inc.php]: You can't remove my copyright, it's fine to add your copyright with it as well but you'll need to restore this line.

File: includes/polling/os/avaya-vsp.inc.php [hunk 5]
```diff
@@ -2,15 +2,29 @@
...
  * option) any later version.  Please see LICENSE.txt at the top level of
  * the source code distribution for details.
  */
 
-$temp     = explode(' ', $poll_device['sysDescr']);
-$hardware = $temp[0];
-$serial   = snmp_get($device, 'rcChasSerialNumber', '-Osqv', 'RAPID-CITY');
-$version  = snmp_get($device, 'rcSysVersion', '-Osqv', 'RAPID-CITY');
+// rcSysVersion
+$version = snmp_get($device, '.1.3.6.1.4.1.2272.1.1.7.0', '-Oqvn');
```
Comment (laf) [inline includes/polling/os/avaya-vsp.inc.php]: Are these just the numerical OID of things like rcChasSerialNumber. If so you should leave them as names so it makes it easier to read.

File: includes/polling/os/avaya-vsp.inc.php [hunk 6]
```diff
@@ -2,15 +2,29 @@
...
+$version = explode(' ', $version);
+$version = $version[0];
+$version = str_replace('"', '', $version);
+
+// rcChasSerialNumber
+$serial = snmp_get($device, '.1.3.6.1.4.1.2272.1.4.2.0', '-Oqvn');
+$serial = str_replace('"', '', $serial);
+
+// rcChasHardwareRevision
+$sysDescr = $poll_device['sysDescr'];
```
Comment (laf) [inline includes/polling/os/avaya-vsp.inc.php]: No point in declaring this. Just use `$poll_device['sysDescr'];` on the line below.

**Comment by laf** [comment]
Nearly all good - just some comments you need to look at and resolve.

File: includes/functions.php [hunk 7]
```diff
@@ -802,7 +802,7 @@ function is_port_valid($port, $device) {
 
     global $config;
 
-    if (strstr($port['ifDescr'], "irtual")) {
+    if (strstr($port['ifDescr'], "irtual") && strpos($port['ifDescr'], "Virtual Services Platform") == false) {
```
Comment (laf) [inline includes/functions.php]: You should do `=== false` here.

**Comment by laf** [comment]
please also see the travis output for some formatting issues:

https://travis-ci.org/librenms/librenms/jobs/155791665

**Comment by ospfbgp** [comment]
This is close to being ready.  I am going to double check everything.  My testing revealed a few issues but now things seem to be working properly on my test vsp switches with discovery and polling.  I would like to call the operating system Avaya VOSS or VOSS in the definitions.inc.php.  Here are some screen shots of what it looks like with Avaya VOSS.
screen1_voss
screen2_voss

**Pull request merged**